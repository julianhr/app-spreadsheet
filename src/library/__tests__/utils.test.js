import {
  getCellNames,
  getColumnLabel,
  getColumnNames,
  getColumnPosition,
  isNumber,
  parseLocation,
  sleep,
} from '../utils'

const columnNames = getColumnNames(800)

describe('#sleep', () => {
  it('awaits on main thread for an amount of milliseconds', async () => {
    const t0 = Date.now()
    await sleep(250)
    const t1 = Date.now()
    const diff = t1 - t0
    expect(diff).toBeGreaterThan(200)
    expect(diff).toBeLessThan(300)
  })
})

describe('#getColumnNames', () => {
  it('returns an array of names', () => {
    expect(columnNames).toMatchSnapshot()
    expect(columnNames.length).toBe(800)
  })
})

describe('#getCellNames', () => {
  it('returns an array of names', () => {
    const columnNames = getCellNames(2, 100)
    expect(columnNames).toMatchSnapshot()
    expect(columnNames.length).toBe(200)
  })
})

describe('#getColumnPosition', () => {
  it('returns position from label', () => {
    for (let i = 0; i < 800; i += 10) {
      const name = columnNames[i]
      expect( getColumnPosition(name) ).toEqual(i)
    }
  })
})

describe('#getColumnLabel', () => {
  it('returns the column name from a position', () => {
    for (let i = 0; i < 800; i += 10) {
      const name = columnNames[i]
      expect( getColumnLabel(i) ).toEqual(name)
    }
  })
})

describe('#parseLocation', () => {
  it('returns an array of row and column as digits', () => {
    const tests = [
      ['A-2', 0, 1],
      ['Z-5', 25, 4],
      ['AA-250', 26, 249],
      ['BC-353', 54, 352],
    ]

    tests.forEach(([ location, row, col ]) => {
      const position = parseLocation(location)
      expect(position).toEqual([row, col])
    })
  })
})

test('#isNumber', () => {
  expect(isNumber('')).toBe(false)
  expect(isNumber('a')).toBe(false)
  expect(isNumber('?')).toBe(false)
  expect(isNumber('.')).toBe(false)
  expect(isNumber('5.5.')).toBe(false)
  expect(isNumber('5.5.5')).toBe(false)
  expect(isNumber('.5.5')).toBe(false)
  expect(isNumber('.5.')).toBe(false)
  expect(isNumber('1')).toBe(true)
  expect(isNumber('000')).toBe(true)
  expect(isNumber('0005')).toBe(true)
  expect(isNumber('5.')).toBe(true)
  expect(isNumber('5.5')).toBe(true)
  expect(isNumber('.5')).toBe(true)
})
