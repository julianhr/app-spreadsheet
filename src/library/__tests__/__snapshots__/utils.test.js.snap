// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#getCellNames returns an array of names 1`] = `
Array [
  "A-1",
  "B-1",
  "C-1",
  "D-1",
  "E-1",
  "F-1",
  "G-1",
  "H-1",
  "I-1",
  "J-1",
  "K-1",
  "L-1",
  "M-1",
  "N-1",
  "O-1",
  "P-1",
  "Q-1",
  "R-1",
  "S-1",
  "T-1",
  "U-1",
  "V-1",
  "W-1",
  "X-1",
  "Y-1",
  "Z-1",
  "AA-1",
  "AB-1",
  "AC-1",
  "AD-1",
  "AE-1",
  "AF-1",
  "AG-1",
  "AH-1",
  "AI-1",
  "AJ-1",
  "AK-1",
  "AL-1",
  "AM-1",
  "AN-1",
  "AO-1",
  "AP-1",
  "AQ-1",
  "AR-1",
  "AS-1",
  "AT-1",
  "AU-1",
  "AV-1",
  "AW-1",
  "AX-1",
  "AY-1",
  "AZ-1",
  "BA-1",
  "BB-1",
  "BC-1",
  "BD-1",
  "BE-1",
  "BF-1",
  "BG-1",
  "BH-1",
  "BI-1",
  "BJ-1",
  "BK-1",
  "BL-1",
  "BM-1",
  "BN-1",
  "BO-1",
  "BP-1",
  "BQ-1",
  "BR-1",
  "BS-1",
  "BT-1",
  "BU-1",
  "BV-1",
  "BW-1",
  "BX-1",
  "BY-1",
  "BZ-1",
  "CA-1",
  "CB-1",
  "CC-1",
  "CD-1",
  "CE-1",
  "CF-1",
  "CG-1",
  "CH-1",
  "CI-1",
  "CJ-1",
  "CK-1",
  "CL-1",
  "CM-1",
  "CN-1",
  "CO-1",
  "CP-1",
  "CQ-1",
  "CR-1",
  "CS-1",
  "CT-1",
  "CU-1",
  "CV-1",
  "A-2",
  "B-2",
  "C-2",
  "D-2",
  "E-2",
  "F-2",
  "G-2",
  "H-2",
  "I-2",
  "J-2",
  "K-2",
  "L-2",
  "M-2",
  "N-2",
  "O-2",
  "P-2",
  "Q-2",
  "R-2",
  "S-2",
  "T-2",
  "U-2",
  "V-2",
  "W-2",
  "X-2",
  "Y-2",
  "Z-2",
  "AA-2",
  "AB-2",
  "AC-2",
  "AD-2",
  "AE-2",
  "AF-2",
  "AG-2",
  "AH-2",
  "AI-2",
  "AJ-2",
  "AK-2",
  "AL-2",
  "AM-2",
  "AN-2",
  "AO-2",
  "AP-2",
  "AQ-2",
  "AR-2",
  "AS-2",
  "AT-2",
  "AU-2",
  "AV-2",
  "AW-2",
  "AX-2",
  "AY-2",
  "AZ-2",
  "BA-2",
  "BB-2",
  "BC-2",
  "BD-2",
  "BE-2",
  "BF-2",
  "BG-2",
  "BH-2",
  "BI-2",
  "BJ-2",
  "BK-2",
  "BL-2",
  "BM-2",
  "BN-2",
  "BO-2",
  "BP-2",
  "BQ-2",
  "BR-2",
  "BS-2",
  "BT-2",
  "BU-2",
  "BV-2",
  "BW-2",
  "BX-2",
  "BY-2",
  "BZ-2",
  "CA-2",
  "CB-2",
  "CC-2",
  "CD-2",
  "CE-2",
  "CF-2",
  "CG-2",
  "CH-2",
  "CI-2",
  "CJ-2",
  "CK-2",
  "CL-2",
  "CM-2",
  "CN-2",
  "CO-2",
  "CP-2",
  "CQ-2",
  "CR-2",
  "CS-2",
  "CT-2",
  "CU-2",
  "CV-2",
]
`;

exports[`#getColumnNames returns an array of names 1`] = `
Array [
  "A",
  "B",
  "C",
  "D",
  "E",
  "F",
  "G",
  "H",
  "I",
  "J",
  "K",
  "L",
  "M",
  "N",
  "O",
  "P",
  "Q",
  "R",
  "S",
  "T",
  "U",
  "V",
  "W",
  "X",
  "Y",
  "Z",
  "AA",
  "AB",
  "AC",
  "AD",
  "AE",
  "AF",
  "AG",
  "AH",
  "AI",
  "AJ",
  "AK",
  "AL",
  "AM",
  "AN",
  "AO",
  "AP",
  "AQ",
  "AR",
  "AS",
  "AT",
  "AU",
  "AV",
  "AW",
  "AX",
  "AY",
  "AZ",
  "BA",
  "BB",
  "BC",
  "BD",
  "BE",
  "BF",
  "BG",
  "BH",
  "BI",
  "BJ",
  "BK",
  "BL",
  "BM",
  "BN",
  "BO",
  "BP",
  "BQ",
  "BR",
  "BS",
  "BT",
  "BU",
  "BV",
  "BW",
  "BX",
  "BY",
  "BZ",
  "CA",
  "CB",
  "CC",
  "CD",
  "CE",
  "CF",
  "CG",
  "CH",
  "CI",
  "CJ",
  "CK",
  "CL",
  "CM",
  "CN",
  "CO",
  "CP",
  "CQ",
  "CR",
  "CS",
  "CT",
  "CU",
  "CV",
  "CW",
  "CX",
  "CY",
  "CZ",
  "DA",
  "DB",
  "DC",
  "DD",
  "DE",
  "DF",
  "DG",
  "DH",
  "DI",
  "DJ",
  "DK",
  "DL",
  "DM",
  "DN",
  "DO",
  "DP",
  "DQ",
  "DR",
  "DS",
  "DT",
  "DU",
  "DV",
  "DW",
  "DX",
  "DY",
  "DZ",
  "EA",
  "EB",
  "EC",
  "ED",
  "EE",
  "EF",
  "EG",
  "EH",
  "EI",
  "EJ",
  "EK",
  "EL",
  "EM",
  "EN",
  "EO",
  "EP",
  "EQ",
  "ER",
  "ES",
  "ET",
  "EU",
  "EV",
  "EW",
  "EX",
  "EY",
  "EZ",
  "FA",
  "FB",
  "FC",
  "FD",
  "FE",
  "FF",
  "FG",
  "FH",
  "FI",
  "FJ",
  "FK",
  "FL",
  "FM",
  "FN",
  "FO",
  "FP",
  "FQ",
  "FR",
  "FS",
  "FT",
  "FU",
  "FV",
  "FW",
  "FX",
  "FY",
  "FZ",
  "GA",
  "GB",
  "GC",
  "GD",
  "GE",
  "GF",
  "GG",
  "GH",
  "GI",
  "GJ",
  "GK",
  "GL",
  "GM",
  "GN",
  "GO",
  "GP",
  "GQ",
  "GR",
  "GS",
  "GT",
  "GU",
  "GV",
  "GW",
  "GX",
  "GY",
  "GZ",
  "HA",
  "HB",
  "HC",
  "HD",
  "HE",
  "HF",
  "HG",
  "HH",
  "HI",
  "HJ",
  "HK",
  "HL",
  "HM",
  "HN",
  "HO",
  "HP",
  "HQ",
  "HR",
  "HS",
  "HT",
  "HU",
  "HV",
  "HW",
  "HX",
  "HY",
  "HZ",
  "IA",
  "IB",
  "IC",
  "ID",
  "IE",
  "IF",
  "IG",
  "IH",
  "II",
  "IJ",
  "IK",
  "IL",
  "IM",
  "IN",
  "IO",
  "IP",
  "IQ",
  "IR",
  "IS",
  "IT",
  "IU",
  "IV",
  "IW",
  "IX",
  "IY",
  "IZ",
  "JA",
  "JB",
  "JC",
  "JD",
  "JE",
  "JF",
  "JG",
  "JH",
  "JI",
  "JJ",
  "JK",
  "JL",
  "JM",
  "JN",
  "JO",
  "JP",
  "JQ",
  "JR",
  "JS",
  "JT",
  "JU",
  "JV",
  "JW",
  "JX",
  "JY",
  "JZ",
  "KA",
  "KB",
  "KC",
  "KD",
  "KE",
  "KF",
  "KG",
  "KH",
  "KI",
  "KJ",
  "KK",
  "KL",
  "KM",
  "KN",
  "KO",
  "KP",
  "KQ",
  "KR",
  "KS",
  "KT",
  "KU",
  "KV",
  "KW",
  "KX",
  "KY",
  "KZ",
  "LA",
  "LB",
  "LC",
  "LD",
  "LE",
  "LF",
  "LG",
  "LH",
  "LI",
  "LJ",
  "LK",
  "LL",
  "LM",
  "LN",
  "LO",
  "LP",
  "LQ",
  "LR",
  "LS",
  "LT",
  "LU",
  "LV",
  "LW",
  "LX",
  "LY",
  "LZ",
  "MA",
  "MB",
  "MC",
  "MD",
  "ME",
  "MF",
  "MG",
  "MH",
  "MI",
  "MJ",
  "MK",
  "ML",
  "MM",
  "MN",
  "MO",
  "MP",
  "MQ",
  "MR",
  "MS",
  "MT",
  "MU",
  "MV",
  "MW",
  "MX",
  "MY",
  "MZ",
  "NA",
  "NB",
  "NC",
  "ND",
  "NE",
  "NF",
  "NG",
  "NH",
  "NI",
  "NJ",
  "NK",
  "NL",
  "NM",
  "NN",
  "NO",
  "NP",
  "NQ",
  "NR",
  "NS",
  "NT",
  "NU",
  "NV",
  "NW",
  "NX",
  "NY",
  "NZ",
  "OA",
  "OB",
  "OC",
  "OD",
  "OE",
  "OF",
  "OG",
  "OH",
  "OI",
  "OJ",
  "OK",
  "OL",
  "OM",
  "ON",
  "OO",
  "OP",
  "OQ",
  "OR",
  "OS",
  "OT",
  "OU",
  "OV",
  "OW",
  "OX",
  "OY",
  "OZ",
  "PA",
  "PB",
  "PC",
  "PD",
  "PE",
  "PF",
  "PG",
  "PH",
  "PI",
  "PJ",
  "PK",
  "PL",
  "PM",
  "PN",
  "PO",
  "PP",
  "PQ",
  "PR",
  "PS",
  "PT",
  "PU",
  "PV",
  "PW",
  "PX",
  "PY",
  "PZ",
  "QA",
  "QB",
  "QC",
  "QD",
  "QE",
  "QF",
  "QG",
  "QH",
  "QI",
  "QJ",
  "QK",
  "QL",
  "QM",
  "QN",
  "QO",
  "QP",
  "QQ",
  "QR",
  "QS",
  "QT",
  "QU",
  "QV",
  "QW",
  "QX",
  "QY",
  "QZ",
  "RA",
  "RB",
  "RC",
  "RD",
  "RE",
  "RF",
  "RG",
  "RH",
  "RI",
  "RJ",
  "RK",
  "RL",
  "RM",
  "RN",
  "RO",
  "RP",
  "RQ",
  "RR",
  "RS",
  "RT",
  "RU",
  "RV",
  "RW",
  "RX",
  "RY",
  "RZ",
  "SA",
  "SB",
  "SC",
  "SD",
  "SE",
  "SF",
  "SG",
  "SH",
  "SI",
  "SJ",
  "SK",
  "SL",
  "SM",
  "SN",
  "SO",
  "SP",
  "SQ",
  "SR",
  "SS",
  "ST",
  "SU",
  "SV",
  "SW",
  "SX",
  "SY",
  "SZ",
  "TA",
  "TB",
  "TC",
  "TD",
  "TE",
  "TF",
  "TG",
  "TH",
  "TI",
  "TJ",
  "TK",
  "TL",
  "TM",
  "TN",
  "TO",
  "TP",
  "TQ",
  "TR",
  "TS",
  "TT",
  "TU",
  "TV",
  "TW",
  "TX",
  "TY",
  "TZ",
  "UA",
  "UB",
  "UC",
  "UD",
  "UE",
  "UF",
  "UG",
  "UH",
  "UI",
  "UJ",
  "UK",
  "UL",
  "UM",
  "UN",
  "UO",
  "UP",
  "UQ",
  "UR",
  "US",
  "UT",
  "UU",
  "UV",
  "UW",
  "UX",
  "UY",
  "UZ",
  "VA",
  "VB",
  "VC",
  "VD",
  "VE",
  "VF",
  "VG",
  "VH",
  "VI",
  "VJ",
  "VK",
  "VL",
  "VM",
  "VN",
  "VO",
  "VP",
  "VQ",
  "VR",
  "VS",
  "VT",
  "VU",
  "VV",
  "VW",
  "VX",
  "VY",
  "VZ",
  "WA",
  "WB",
  "WC",
  "WD",
  "WE",
  "WF",
  "WG",
  "WH",
  "WI",
  "WJ",
  "WK",
  "WL",
  "WM",
  "WN",
  "WO",
  "WP",
  "WQ",
  "WR",
  "WS",
  "WT",
  "WU",
  "WV",
  "WW",
  "WX",
  "WY",
  "WZ",
  "XA",
  "XB",
  "XC",
  "XD",
  "XE",
  "XF",
  "XG",
  "XH",
  "XI",
  "XJ",
  "XK",
  "XL",
  "XM",
  "XN",
  "XO",
  "XP",
  "XQ",
  "XR",
  "XS",
  "XT",
  "XU",
  "XV",
  "XW",
  "XX",
  "XY",
  "XZ",
  "YA",
  "YB",
  "YC",
  "YD",
  "YE",
  "YF",
  "YG",
  "YH",
  "YI",
  "YJ",
  "YK",
  "YL",
  "YM",
  "YN",
  "YO",
  "YP",
  "YQ",
  "YR",
  "YS",
  "YT",
  "YU",
  "YV",
  "YW",
  "YX",
  "YY",
  "YZ",
  "ZA",
  "ZB",
  "ZC",
  "ZD",
  "ZE",
  "ZF",
  "ZG",
  "ZH",
  "ZI",
  "ZJ",
  "ZK",
  "ZL",
  "ZM",
  "ZN",
  "ZO",
  "ZP",
  "ZQ",
  "ZR",
  "ZS",
  "ZT",
  "ZU",
  "ZV",
  "ZW",
  "ZX",
  "ZY",
  "ZZ",
  "AAA",
  "AAB",
  "AAC",
  "AAD",
  "AAE",
  "AAF",
  "AAG",
  "AAH",
  "AAI",
  "AAJ",
  "AAK",
  "AAL",
  "AAM",
  "AAN",
  "AAO",
  "AAP",
  "AAQ",
  "AAR",
  "AAS",
  "AAT",
  "AAU",
  "AAV",
  "AAW",
  "AAX",
  "AAY",
  "AAZ",
  "ABA",
  "ABB",
  "ABC",
  "ABD",
  "ABE",
  "ABF",
  "ABG",
  "ABH",
  "ABI",
  "ABJ",
  "ABK",
  "ABL",
  "ABM",
  "ABN",
  "ABO",
  "ABP",
  "ABQ",
  "ABR",
  "ABS",
  "ABT",
  "ABU",
  "ABV",
  "ABW",
  "ABX",
  "ABY",
  "ABZ",
  "ACA",
  "ACB",
  "ACC",
  "ACD",
  "ACE",
  "ACF",
  "ACG",
  "ACH",
  "ACI",
  "ACJ",
  "ACK",
  "ACL",
  "ACM",
  "ACN",
  "ACO",
  "ACP",
  "ACQ",
  "ACR",
  "ACS",
  "ACT",
  "ACU",
  "ACV",
  "ACW",
  "ACX",
  "ACY",
  "ACZ",
  "ADA",
  "ADB",
  "ADC",
  "ADD",
  "ADE",
  "ADF",
  "ADG",
  "ADH",
  "ADI",
  "ADJ",
  "ADK",
  "ADL",
  "ADM",
  "ADN",
  "ADO",
  "ADP",
  "ADQ",
  "ADR",
  "ADS",
  "ADT",
]
`;
